### PR TITLE
Insertion sort for Kotlin added

### PR DESCRIPTION
**Pull request by gat786**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes, I have.

### Description

This commit includes a Insertion sort in the kotlin language corpus of algorithms

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

It has no related issues as per my information

### CODE REVIEW

File: Kotlin/sort/InsertionSort/src/InsertionSort.kt [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    }
+    return array
+}
+
+fun main() {
+    val list = listOf(11,22,33,45,34,53,27,36)
+    println("unsorted list is $list")
+    val sorted = insertionSort(list.toMutableList());
+    println("sorted list is $sorted")
+}
```
Comment (HarshCasper) [inline Kotlin/sort/InsertionSort/src/InsertionSort.kt]: Add a new line at the EOF

File: Kotlin/sort/InsertionSort/src/InsertionSort.kt [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        // when the loop has ended we know that the value at index
+        // less then current temp is lesser than current value so
+        // replace value at temp and close the loop
+        array[temp] = value
+    }
+    return array
+}
+
+fun main() {
+    val list = listOf(11,22,33,45,34,53,27,36)
```
Comment (HarshCasper) [inline Kotlin/sort/InsertionSort/src/InsertionSort.kt]: Make the input dynamic

File: Kotlin/sort/InsertionSort/src/InsertionSort.kt [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        // if the value is not greater than the loop will end
+        while ((array[temp - 1] > value) and (temp >= 0)) {
+            // put new_value one index above the current value
+            array[temp] = array[temp - 1]
+            // decrease the temporary index
+            temp -= 1
+        }
+        // when the loop has ended we know that the value at index
+        // less then current temp is lesser than current value so
+        // replace value at temp and close the loop
```
Comment (HarshCasper) [inline Kotlin/sort/InsertionSort/src/InsertionSort.kt]: Use Multi-Line Comments here

File: Kotlin/sort/InsertionSort/src/InsertionSort.kt [hunk 4]
```diff
@@ -0,0 +1,32 @@
+fun insertionSort(array: MutableList<Int>) : List<Int> {
+    // go from the 2nd item in the list till the last item
+    for (i in 1 until array.size){
+        // store the value
+        val value = array[i]
+        // store the index of the value
+        var temp = i
+
+        // for every new_value below the index of the value
+        // if the new_value is greater than the value
```
Comment (HarshCasper) [inline Kotlin/sort/InsertionSort/src/InsertionSort.kt]: Use Multi-Line Comments here

**Review by HarshCasper — CHANGES_REQUESTED** [review]

File: Kotlin/sort/InsertionSort/src/InsertionSort.kt (hunk 3 above)
Comment (gat786) [reply Kotlin/sort/InsertionSort/src/InsertionSort.kt]: okay

File: Kotlin/sort/InsertionSort/src/InsertionSort.kt (hunk 4 above)
Comment (gat786) [reply Kotlin/sort/InsertionSort/src/InsertionSort.kt]: okay

File: Kotlin/sort/InsertionSort/src/InsertionSort.kt (hunk 1 above)
Comment (gat786) [reply Kotlin/sort/InsertionSort/src/InsertionSort.kt]: okay

File: Kotlin/sort/InsertionSort/src/InsertionSort.kt (hunk 2 above)
Comment (gat786) [reply Kotlin/sort/InsertionSort/src/InsertionSort.kt]: Do you mean to make it so that it accepts inputs from `stdin`

Comment (gat786) [reply Kotlin/sort/InsertionSort/src/InsertionSort.kt]: @HarshCasper  i added the requested changes in a commit and pushed it. let me know if it is okay.

**Pull request merged**